### PR TITLE
Fix organisation workspace list

### PR DESCRIPTION
**Pull request by MatthewJohn**
# NOTE!!

As per the following notes, I do not have access to any Terraform enterprise/cloud instances - the changes are based:
 - The pre-existing JSON being invalid
 - Updating to be consistent with other APIs that provide similar style responses.

If someone from Hashicorp could review the proposed change against the actual schema of the response please :)

### What
Fix API response for Workspace list endpoint.

### Why
This was previously documented to return an object, which did not contain keys. Have updated to return a list of objects from the root 'data' object, based on the API outputs from other APIs that return a list of objects.

### Screenshots

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [ ] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc). (I do not have permissions to add labels)
- [x] Description links to related pull requests or issues, if any. (I have not created an issue, but can do if need be, otherwise N/A)

#### Content
- [x] Redirects have been added for moved, renamed, or deleted pages. This requires a separate PR in the [`terraform-website` repository](https://github.com/hashicorp/terraform-website) `redirects.next.js` file. (I don't think this applies - N/A)
- [x] API documentation and the API Changelog have been updated.  (The API documentation has been corrected)
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Pages with related content are updated and link to this content when appropriate.
- [x] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages. (N/A)
- [x] New pages have metadata (page name and description) at the top. (N/A)
- [x] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility. (N/A)
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] UI elements (button names, page names, etc.) are bolded. (N/A)
- [ ] The Vercel website preview successfully deployed.

#### Reviews
- [ ] I or someone else reviewed the content for technical accuracy.
- [ ] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

**Comment by mgarrell777** [comment]
Hi @MatthewJohn Unfortunately, your Vercel build is failing currently. However, this issue should be a 1 time thing. If you update your branch with the latest main, that should resolve the error.

**Comment by MatthewJohn** [comment]
> Hi @MatthewJohn Unfortunately, your Vercel build is failing currently. However, this issue should be a 1 time thing. If you update your branch with the latest main, that should resolve the error.

Thanks Matt, Just amended commit and will see if it works this time :crossed_fingers:

**Comment by thiskevinwang** [comment]
Hey @MatthewJohn, if you hit the **Update branch** option (or rebase manually on origin/main), that should resolve the Vercel build errors! 

(There are some recently added files in `main` that aren't in your fork!_

**Comment by MatthewJohn** [comment]
> Hey @MatthewJohn, if you hit the **Update branch** option (or rebase manually on origin/main), that should resolve the Vercel build errors!
> 
> (There are some recently added files in `main` that aren't in your fork!_

Ah, makes sense :) All done

**Comment by MatthewJohn** [comment]
> Hi @MatthewJohn Unfortunately, your Vercel build is failing currently. However, this issue should be a 1 time thing. If you update your branch with the latest main, that should resolve the error.

Doh, I had misread what you'd said and thought you'd just said 'update your branch', in the sense of 'make some change to trigger CI again', thining the '1 time thing' was a flaky build :doh:

File: website/docs/cloud-docs/api-docs/workspaces.mdx [hunk 1]
```diff
@@ -675,120 +675,120 @@ $ curl \
...
+        "queue-all-runs": false,
+        "resource-count": 0,
+        "run-failures": null,
+        "source": "tfe-api",
+        "source-name": null,
+        "source-url": null,
+        "speculative-enabled": true,
+        "structured-run-output-enabled": true,
+        "terraform-version": "0.11.1",
+        "trigger-prefixes": [],
```
Comment (matejrisek) [inline website/docs/cloud-docs/api-docs/workspaces.mdx:726]: we might add `"trigger-patterns": [],` since we're returning it

**Review by matejrisek — COMMENTED** [review]
@MatthewJohn thanks for the contribution 🎉 🚀 

@mgarrell777 @thiskevinwang we have other possible mismatched/outdated responses for workspaces => we should probably address them all in one go.

**Comment by MatthewJohn** [comment]
@matejrisek , not sure if it's related to the outdated response, but I've also raised https://github.com/hashicorp/terraform-docs-common/pull/85 for a workspace response issue :)

**Review by mgarrell777 — APPROVED** [review]
Hi @MatthewJohn Thanks very much for your contribution Looks good to me.

**Comment by mgarrell777** [comment]
@MatthewJohn Do you still want to merge this PR? If so, it looks like there's a merge conflict to resolve. I can merge it once that resolution occurs.

**Comment by MatthewJohn** [comment]
@mgarrell777 Yes please, I'll resolve tonight :)

**Pull request merged**